addon/-private/system/record-arrays/immutable-array-mixin.js

```diff
@@ -0,0 +1,34 @@
+import Ember from 'ember';
+import isEnabled from 'ember-data/-private/features';
+
+const { get } = Ember;
+
+const MUTATION_METHODS = [
+  'addObject',
+  'addObjects',
+  'removeObject',
+  'removeObjects',
+  'unshiftObject',
+  'unshiftObjects',
+  'pushObject',
+  'pushObjects',
+  'reverseObjects',
+  'setObjects',
+  'shiftObject'
```

Comment:
Should the following methods be added as well, since they are public and mutate the array too?

- [`clear`](http://emberjs.com/api/classes/Ember.MutableArray.html#method_clear)
- [`popObject`](http://emberjs.com/api/classes/Ember.MutableArray.html#method_popObject)
- [`removeAt`](http://emberjs.com/api/classes/Ember.MutableArray.html#method_removeAt)
- [`removeObject`](http://emberjs.com/api/classes/Ember.MutableArray.html#method_removeObject)
- [`removeObjects`](http://emberjs.com/api/classes/Ember.MutableArray.html#method_removeObjects)
- [`insertAt`](http://emberjs.com/api/classes/Ember.MutableArray.html#method_insertAt)